File: clojure/word-count/word_count.clj
(ns word-count)

(defn extract-words [words]
  (re-seq #"\w+" words))

(defn word-frequency [col key]
  (inc (col key 0)))

(defn word-count [words]
  (let [words (clojure.string/lower-case words)]
    (reduce #(assoc %1 %2 (word-frequency %1 %2)) {} (extract-words words))))

File: javascript/word-count/word-count.js
module.exports = function (phrase) {
  var words = phrase.split(/[\s\n]/);

  return words.reduce(function (count, word) {
    if (count[word]) {
      count[word]++;
    } else {
      count[word] = 1;
    }

    return count;
  }, {});
}
